### PR TITLE
[TEST] check both old and new derivative suffix

### PR DESCRIPTION
**Pull request by htwangtw**
Add test to verify this note in comments:
 
    fmriprep has changed the file suffix between v20.1.1 and v20.2.0 with
    respect to BEP 012.
    cf. https://neurostars.org/t/naming-change-confounds-regressors-to-confounds-timeseries/17637 # noqa
    Check file with new naming scheme exists or replace,
    for backward compatibility.

This test was lost while migrating the library.
Although it didn't break any existing code, it's still better to have this test.

### CODE REVIEW

File: nilearn/interfaces/fmriprep/tests/test_load_confounds.py [hunk 1]
```diff
@@ -423,11 +423,12 @@ def test_load_non_nifti(tmp_path):
 
 def test_invalid_filetype(tmp_path):
     """Invalid file types/associated files for load method."""
-    bad_nii, bad_conf = create_tmp_filepath(tmp_path, copy_confounds=True)
+    bad_nii, bad_conf = create_tmp_filepath(tmp_path, copy_confounds=True,
+                                            old_deriveative_suffix=False)
```
Comment (bthirion) [inline nilearn/interfaces/fmriprep/tests/test_load_confounds.py:427]:
```suggestion
                                            old_derivative_suffix=False)
```

File: nilearn/interfaces/fmriprep/tests/utils.py [hunk 2]
```diff
@@ -47,10 +48,13 @@ def create_tmp_filepath(
     suffix="test",
     copy_confounds=False,
     copy_json=False,
+    old_deriveative_suffix=False
```
Comment (bthirion) [inline nilearn/interfaces/fmriprep/tests/utils.py:51]:
```suggestion
    old_derivative_suffix=False
```

**Review by bthirion — COMMENTED** [review]
Idea is OK, just a typo.

File: nilearn/interfaces/fmriprep/tests/utils.py [hunk 3]
```diff
@@ -47,10 +48,13 @@ def create_tmp_filepath(
     suffix="test",
     copy_confounds=False,
     copy_json=False,
+    old_derivative_suffix=False
 ):
     """Create test files in temporary directory."""
+    deriveative = "regressors" if old_derivative_suffix else "timeseries"
```
Comment (NicolasGensollen) [inline nilearn/interfaces/fmriprep/tests/utils.py:54]:
```suggestion
    derivative = "regressors" if old_derivative_suffix else "timeseries"
```

File: nilearn/interfaces/fmriprep/tests/utils.py [hunk 4]
```diff
@@ -47,10 +48,13 @@ def create_tmp_filepath(
...
+
     # confound files
-    confounds_root = "_desc-confounds_regressors.tsv"
+    confounds_root = f"_desc-confounds_{deriveative}.tsv"
```
Comment (NicolasGensollen) [inline nilearn/interfaces/fmriprep/tests/utils.py:57]:
```suggestion
    confounds_root = f"_desc-confounds_{derivative}.tsv"
```

File: nilearn/interfaces/fmriprep/tests/utils.py [hunk 5]
```diff
@@ -60,7 +64,7 @@ def create_tmp_filepath(
         tmp_conf.touch()
 
     if copy_json:
-        meta_root = "_desc-confounds_regressors.json"
+        meta_root = f"_desc-confounds_{deriveative}.json"
```
Comment (NicolasGensollen) [inline nilearn/interfaces/fmriprep/tests/utils.py:67]:
```suggestion
        meta_root = f"_desc-confounds_{derivative}.json"
```

**Review by NicolasGensollen — COMMENTED** [review]
Thanks @htwangtw !
Just a few typos:

**Review by NicolasGensollen — APPROVED** [review]
LGTM, thanks!

**Review by bthirion — COMMENTED** [review]
LGTM !

**Review by bthirion — APPROVED** [review]

**Comment by NicolasGensollen** [comment]
Thanks! Merging...

**Pull request merged**